packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/customization/implementation/ls/EclipseLanguageServerFacade.java

```diff
@@ -104,9 +144,12 @@ private static Path getLanguageServerDirectory(int javaVersion, Logger logger) t
         }
 
         Path languageServer = languageServerPath.resolve("jdt-language-server");
-        if (!Files.exists(languageServerPath) || !Files.exists(languageServer)) {
-            Files.createDirectories(languageServerPath);
+
+        if (!Files.exists(languageServer) || forceReDownload) {
+            Files.createDirectories(languageServer);
```

Comment:
Would this `createDirectories` clean the languageServer, on case of forceReDownload=true?

Or it waits for `unzipLanguageServer` method?

Reply:
No, it'll do nothing if the folder exists.
Also, `Files.deleteIfExists` will only delete folder if it's empty. To clean the folder, we have to walk the file tree and delete files one by one.
Current implementation is for unzip to clean existing files that exists in the tar.gz file.

Reply:
OK, so we didn't clean up the folder. Just the unzip would overwrite the files in the folder.

Reply:
Yes.